### ISSUE TITLE
Incomplete header found

### ISSUE THREAD

**Issue by minborg** [OpenHFT/Chronicle-Queue#720]
In test `SingleChronicleQueueTest` there is an exception:

```
java.lang.Exception: Incomplete header found at pos: 524572: c0000000, overwriting
	at net.openhft.chronicle.wire.AbstractWire.enterHeader(AbstractWire.java:312)
	at net.openhft.chronicle.queue.impl.single.StoreAppender.writeHeader(StoreAppender.java:403)
	at net.openhft.chronicle.queue.impl.single.StoreAppender.openContext(StoreAppender.java:408)
	at net.openhft.chronicle.queue.impl.single.StoreAppender.writingDocument(StoreAppender.java:348)
	at net.openhft.chronicle.queue.impl.single.StoreAppender.writingDocument(StoreAppender.java:315)
	at net.openhft.chronicle.wire.MarshallableOut.writeText(MarshallableOut.java:158)
	at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueTest.lambda$testTailerSnappingRollWithNewAppender$91(SingleChronicleQueueTest.java:2684)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at net.openhft.chronicle.core.threads.CleaningThread.run(CleaningThread.java:38)
```

**Comment by peter-lawrey** [comment]
I have seen this as well, but only when running all the tests in the class.

**Comment by RobAustin** [comment]
@minborg can you outline steps to reproduce or close the issue as won't fix.

**Comment by minborg** [comment]
It's still there:

```
java.lang.Exception: Incomplete header found at pos: 524572: c0000000, overwriting
	at net.openhft.chronicle.wire.AbstractWire.enterHeader(AbstractWire.java:312)
	at net.openhft.chronicle.queue.impl.single.StoreAppender.writeHeader(StoreAppender.java:453)
	at net.openhft.chronicle.queue.impl.single.StoreAppender.openContext(StoreAppender.java:458)
	at net.openhft.chronicle.queue.impl.single.StoreAppender.writingDocument(StoreAppender.java:399)
	at net.openhft.chronicle.queue.impl.single.StoreAppender.writingDocument(StoreAppender.java:358)
	at net.openhft.chronicle.wire.MarshallableOut.writeText(MarshallableOut.java:158)
	at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueTest.lambda$testTailerSnappingRollWithNewAppender$91(SingleChronicleQueueTest.java:2831)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at net.openhft.chronicle.core.threads.CleaningThread.run(CleaningThread.java:67)

```

**Comment by minborg** [comment]
The problem is related to the parameters `wireType=BINARY_LIGTH` and `encrypted=false` and `testTailerSnappingRollWithNewAppender()`

**Comment by minborg** [comment]
This was related to a missing time provider in the actual test

**Comment by JerryShea** [comment]
@minborg how come the missing time provider gave us such an obscure error?

**Comment by minborg** [comment]
The test was set up with three queues pointing to the same directory. The middle queue didn't have a custom time provider and so, messages were inserted out of time order.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.20.123](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.20.123), [BOM-2.20.226](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.226)
